File: 338.-Counting-Bits/counting_bits_v1.py
class Solution(object):

	def countBits(self, num):
	    """
	    :type num: int
	    :rtype: List[int]
	    """
	    list1 = []

	    for n in range(0, num + 1):
	        tmp = n
	        occur = 0
	        list2 = []
	        while (tmp != 0):
	            rmd = tmp % 2
	            tmp = (tmp - rmd) / 2
	            list2.append(rmd)
	            
	            
	        for m in range(0, len(list2)):
	            if list2[m] == 1:
	                occur = occur + 1

	        list1.append(occur)

	    return list1

File: 338.-Counting-Bits/counting_bits_v1.cpp
#include <iostream>
#include <vector>

using namespace std;

vector <int> countBits(int num) {
    vector <int> result;

	for (int i = 0; i <= num; i++)
	{
		int tmp = i;
		int occur = 0;
		while (tmp != 0)
		{
			int rmd = tmp % 2;
			tmp = (tmp - rmd) / 2;
			if (rmd == 1) occur++;
		}
		result.push_back(occur);
	}
    return result;
}

int main(int argc, char const *argv[])
{
	vector <int> test;
	test = countBits(5);
	cout << "[";
	for (unsigned int i = 0; i < test.size() - 1; i++)
	{
		cout << test[i] << " ";
	}
	cout << test[test.size() - 1];
	cout << "]" << endl;
	return 0;
}
